src/main/java/com/ndgl/spotfinder/domain/post/service/PostViewCountService.java

```diff
@@ -0,0 +1,94 @@
+package com.ndgl.spotfinder.domain.post.service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.springframework.data.redis.connection.DataType;
+import org.springframework.data.redis.connection.RedisConnection;
+import org.springframework.data.redis.core.Cursor;
+import org.springframework.data.redis.core.KeyScanOptions;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.ScanOptions;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.ndgl.spotfinder.domain.post.entity.Post;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class PostViewCountService {
+	private final RedisTemplate<String, String> redisTemplate;
+	private final PostService postService;
+
+	private static final String POST_KEY_PREFIX = "viewed:post:";
+	private static final long SCAN_BATCH_SIZE = 1000;
+
+	public Map<Long, Long> collectPostViewCounts() {
+		Map<Long, Long> viewCountMap = new HashMap<>();
+		RedisConnection connection = Objects.requireNonNull(redisTemplate.getConnectionFactory()).getConnection();
+
+		KeyScanOptions options = (KeyScanOptions)ScanOptions
+			.scanOptions()
+			.type(DataType.SET)
+			.match(POST_KEY_PREFIX + "*")
+			.count(SCAN_BATCH_SIZE)
+			.build();
+
+		try (Cursor<byte[]> cursor = connection.scan(options)) {
+			while (cursor.hasNext()) {
+				String key = new String(cursor.next());
+				Long postId = extractPostIdFromKey(key);
+				Long viewCount = redisTemplate.opsForSet().size(key);
+
+				viewCountMap.put(postId, viewCount);
+			}
+		} catch (Exception e) {
+			log.error("조회수를 가져오지 못했습니다.");
+		}
+
+		return viewCountMap;
+	}
+
+	@Transactional
+	public void applyToDatabase(Map<Long, Long> viewCounts) {
+		List<Long> postIds = viewCounts.keySet().stream().toList();
+		List<Post> posts = postService.findPostByIds(postIds);
+
+		posts.forEach(post -> {
+			post.addViewCount(viewCounts.get(post.getId()));
+		});
```

Comment:
조회한 후에 업데이트하고 다시 저장하는 로직 대신 바로 업데이트 쿼리를 날리는 건 어떠실까요? 의견 궁금합니다 !

- PostRepository
```java
@Modifying
@Query("UPDATE Post p SET p.viewCount = p.viewCount + :count WHERE p.id = :postId")
void incrementViewCount(@Param("postId") Long postId, @Param("count") Long count);
```

- 로직 수정
```suggestion
		 viewCounts.forEach((postId, viewCount) -> {
        postRepository.incrementViewCount(postId, viewCount);
    });
```

Reply:
포스트가 많아진다면 UPDATE문만 사용하는 것이 좀 더 효율적일 것 같네요. 이 부분은 수정해보겠습니다.